secp256k1jni/src/main/java/org/bitcoin/NativeSecp256k1.java

```diff
@@ -494,6 +494,80 @@ public static boolean isValidPubKey(byte[] pubkey) {
         return retVal == 0 ? new byte[0] : sigArr;
     }
 
+    /**
+     * libsecp256k1 Compute pubkey associated with schnorr nonce.
+     *
+     * @param nonce Schnorr nonce, 32 bytes
+     * @return pubkey 33 byte ECPublicKey associated with nonce
+     */
+    public static byte[] schnorrPublicNonce(byte[] nonce) throws AssertFailException {
+        checkArgument(nonce.length == 32);
+
+        ByteBuffer byteBuff = nativeECDSABuffer.get();
+        if (byteBuff == null || byteBuff.capacity() < nonce.length) {
+            byteBuff = ByteBuffer.allocateDirect(32);
+
+            nativeECDSABuffer.set(byteBuff);
+        }
+        byteBuff.rewind();
+        byteBuff.put(nonce);
+
+        byte[][] retByteArray;
+
+        r.lock();
+        try {
+            retByteArray = secp256k1_schnorrsig_pubnonce(byteBuff, Secp256k1Context.getContext());
+        } finally {
+            r.unlock();
+        }
+
+        byte[] pubKeyArr = retByteArray[0];
+        int retVal = new BigInteger(new byte[] { retByteArray[1][0] }).intValue();
+
+        assertEquals(pubKeyArr.length, 33, "Got bad pubkey length.");
+
+        return retVal == 0 ? new byte[0] : pubKeyArr;
+    }
+
+    /**
+     * libsecp256k1 Create a Schnorr signature using a specified nonce.
+     *
+     * @param data Message hash, 32 bytes
+     * @param seckey Secret key, 32 bytes
+     * @param nonce One-time nonce, 32 bytes
+     * @return sig byte array of signature
+     */
+    public static byte[] schnorrSignWithNonce(byte[] data, byte[] seckey, byte[] nonce) throws AssertFailException {
+        checkArgument(data.length == 32 && seckey.length <= 32 && nonce.length == 32);
+
+        ByteBuffer byteBuff = nativeECDSABuffer.get();
+        if (byteBuff == null || byteBuff.capacity() < data.length + seckey.length + nonce.length) {
+            byteBuff = ByteBuffer.allocateDirect(32 + 32 + 32);
```

Comment:
I think this should be `seckey.length` instead of `32` as we state that `seckey.length <= 32`

Reply:
changed the requirement to `seckey.length == 32` instead